### PR TITLE
Add license friendly JDBC drivers into server-core

### LINKED ISSUE #732: Bundle license-friendly JDBC drivers
As a developer, I'd like to add license-friendly JDBC drivers as part of the core dependencies, so it is easy for end users to connect to the desired RDBMS. For other database types, we would want to have it documented describing the procedure.

**Acceptance:**
- Following drivers are bundled: PostgreSQL (650KB), MySQL (Maria DB driver, 358K), H2 (already included as default) and HSQLDB (1.5MB)
- Docs describe the procedure how to add other driver types

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add MariaDB java client (that supports MariaDB, MySQL), PostgreSQL and HSQLDB drivers
 - Add doc that explains how to configure RDBMS

This resolves #732

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -149,6 +149,36 @@ to `WARNING` in `logback.xml`:
 
 ====
 
+[[configuring-rdbms]]
+== RDBMS configuration
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: This needs to be part of other server implementations as well. may be we can split this part into a separate doc and have it added in each server implementations?

**Comment by ilayaperumalg** [comment]
I have verified connecting to MySQL (using MariaDB driver), PostgreSQL, HSQLDB with these driver additions.

File: spring-cloud-dataflow-server-core/pom.xml [hunk 2]
```diff
@@ -119,6 +119,21 @@
 			<artifactId>h2</artifactId>
 		</dependency>
 		<dependency>
+			<groupId>org.hsqldb</groupId>
+			<artifactId>hsqldb</artifactId>
+			<version>${hsqldb.version}</version>
```
Comment (trisberg) [inline spring-cloud-dataflow-server-core/pom.xml]: do we need the <version> tag here?

Comment (cppwfs) [inline spring-cloud-dataflow-server-core/pom.xml]: They can be removed.

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc [hunk 3]
```diff
@@ -149,6 +149,36 @@ to `WARNING` in `logback.xml`:
...
+If you are using any other RDBMS, then the corresponding JDBC driver jar needs to be on the classpath of the server.
+
+The RDBMS properties can be passed as command-line arguments to the dataflow server.
+
+For instance,
+If you are using MySQL,
+
+```
+java -jar spring-cloud-dataflow-server-local/target/spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar \
+ --spring.datasource.url=jdbc:mysql:<db-info> --spring.datasource.username=<user> --spring.datasource.password=<password> --spring.datasource.driver-class-name=org.mariadb.jdbc.Driver &
```
Comment (cppwfs) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: the datasource should be `jdbc:mariadb:`

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc (hunk 1 above)
Comment (trisberg) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: I think the configuration would be different for different server implementations, CF adds it's own configuration and so does k8s, so switching DB would not be the same instructions across the board

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc (hunk 3 above)
Comment (ilayaperumalg) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: In this case, we are using just the MariaDB JDBC driver to connect to the MySQL server. Hence, the `url` still points to `jdbc:mysql`.

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc (hunk 1 above)
Comment (ilayaperumalg) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: ok, I see. @trisberg thank you for clarifying this.

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc (hunk 3 above)
Comment (cppwfs) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: coolness

**Comment by ghillert** [comment]
Merged with commit 66fc696.

Did some Asciidoc polishing and made sure that we use the naming `Data Flow Server`.

**Pull request closed without merging**